src/oci.c

```diff
@@ -236,13 +236,9 @@ cc_oci_kill (struct cc_oci_config *config,
 		goto error;
 	}
 
-#ifndef UNIT_TESTING
-	if (! cc_proxy_hyper_kill_container(config, signum)) {
-		g_critical("failed to kill container");
-		goto error;
-	}
-#else
-	//FIXME: should we kill to cc-shim?
+	/* send signal to cc-shim because it
+	 * will send killcontainer to hyperstart
+	 */
 	if (kill (state->pid, signum) < 0) {
```

Comment:
This is fine for most circumstances, but not if `signum == SIGKILL` as the shim won't be able to handle that and will be killed immediately, leaving the container running.

We need a special case so that if `signum == SIGKILL`, the shim is killed and `cc_proxy_hyper_kill_container()` is also called to shutdown the container.